### PR TITLE
Refactor: Move doc gen from sparrow binary to its own script

### PR DESCRIPTION
**Pull request by NiklasTreml**
## Motivation

The final sparrow binary should not include the gen-docs command, as it is only intended to generate the docs for the repo.
## Changes

Moved the gen-docs into its own package under `scripts/gen-docs`. It can be invoked through `go generate ./...` as before. 

For additional information look at the commits.

## Tests done

- [ ] Regenerated documentation, `git diff` shows same output

### CODE REVIEW

File: docs/sparrow.md [hunk 1]
```diff
@@ -16,7 +16,6 @@ The check results are exposed via an API.
 
 ### SEE ALSO
 
-* [sparrow completion](sparrow_completion.md)	 - Generate the autocompletion script for the specified shell
-* [sparrow gen-docs](sparrow_gen-docs.md)	 - Generate markdown documentation
 * [sparrow run](sparrow_run.md)	 - Run sparrow
 
+###### Auto generated by spf13/cobra on 10-Jan-2024
```
Comment (lvlcn-t) [inline docs/sparrow.md:21]: Disable this auto gen tag since the pre-commit hook would always fail once a day.

File: docs/sparrow_run.md [hunk 2]
```diff
@@ -37,3 +37,4 @@ sparrow run [flags]
 
 * [sparrow](sparrow.md)	 - Sparrow, the infrastructure monitoring agent
 
+###### Auto generated by spf13/cobra on 10-Jan-2024
```
Comment (lvlcn-t) [inline docs/sparrow_run.md:40]: Disable this auto gen tag since the pre-commit hook would always fail once a day.

**Review by lvlcn-t — CHANGES_REQUESTED** [review]
One minor thing but after it is fixed: LGTM

**Review by puffitos — COMMENTED** [review]
Let's remove the date on the auto generation - we can check the date on the modified date of the file :)

File: scripts/gen-docs/gen-docs.go [hunk 3]
```diff
@@ -42,9 +62,10 @@ func NewCmdGenDocs(rootCmd *cobra.Command) *cobra.Command {
...
 
 // runGenDocs generates the markdown files for the flag documentation
-func runGenDocs(rootCmd *cobra.Command, path *string) func(cmd *cobra.Command, args []string) {
-	rootCmd.DisableAutoGenTag = true
+func runGenDocs(path *string) func(cmd *cobra.Command, args []string) {
+	irgendwascmd := sparrowcmd.BuildCmd("")
+	irgendwascmd.DisableAutoGenTag = true
 	return func(cmd *cobra.Command, args []string) {
-		_ = doc.GenMarkdownTree(rootCmd, *path)
+		_ = doc.GenMarkdownTree(irgendwascmd, *path)
```
Comment (puffitos) [inline scripts/gen-docs/gen-docs.go:69]: I would check for the error, we should still check if something goes wrong during doc creation. `irgendwascmd` is funny and I do appreciate a funny name, but I think given the scope of the variable a simple `c` would be enough :)

Comment (NiklasTreml) [reply scripts/gen-docs/gen-docs.go:70]: oh yeah forgot to change that :D

**Review by lvlcn-t — APPROVED** [review]

**Review by y-eight — APPROVED** [review]
LGTM

**Pull request merged**